src/Symfony/Component/Mailer/Bridge/Mailgun/Transport/MailgunHttpTransport.php

```diff
@@ -66,13 +67,18 @@ protected function doSendHttp(SentMessage $message): ResponseInterface
             'body' => $body->bodyToIterable(),
         ]);
 
-        $result = $response->toArray(false);
+        try {
+            $result = $response->toArray(false);
+        } catch (TransportExceptionInterface $e) {
+            throw new HttpTransportException('Could not reach the remote Mailgun server.', $response, 0, $e);
+        }
+
         if (200 !== $response->getStatusCode()) {
```

Comment:
I'd wrap the call to `getStatusCode` and `toArray` in the same try/catch (same everywhere)

Reply:
Sure. I will but why? This will never throw `TransportExceptionInterface` if I first get the body.

Reply:
I'd say to make things easy to reason about and consistent (some places were calling getStatusCode before toArray/getContent)